packages/demo/src/components/ActionBox/ActionBox.tsx

```diff
@@ -0,0 +1,66 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import { useState, MouseEvent as ReactMouseEvent, useCallback } from "react";
+import { CopyBlock, irBlack } from "react-code-blocks";
+
+type actionFnRequiredParam<T = unknown, U = unknown> = (option: T) => U;
+type actionFnOptionalParam<T = unknown, U = unknown> = (option?: T) => U;
+
+interface IActionBox<T = unknown, U = unknown> {
+  actionTitle: string;
+  actionFn: actionFnOptionalParam<T, U> | actionFnRequiredParam<T, U>;
+  actionOption: T;
+  actionCode: string;
+  actionFnTestId?: string;
+}
```

Comment:
I'd have to keep the different action fn types `required` and `optional` because there are functions which require a required params like `connect`, and `addVerifiableCredentialRequest`.

Reply:
I think you can try this signature for the method. `(option: T | undefined) => U`. Maybe it'll work.

Reply:
```
Type '(proofType: MerkleProofType) => void' is not assignable to type '(option: MerkleProofType | undefined) => void'.ts(2322)
```

That error for example came from: 
```
<ActionBox<MerkleProofType, void>
      code={RLN_CODE}
      onClick={genRLNProof}
      option={MerkleProofType.ARTIFACTS}
      title="Generate proof from Merkle proof artifacts"
    />
```
We would need separate function signatures for different cases.

Reply:
Ok, let's leave two methods instead.